src/BuildIt.Forms/BuildIt.Forms.Controls/Platforms/Android/AutoFitTextureView.cs

```diff
@@ -45,17 +48,45 @@ protected override void OnMeasure(int widthMeasureSpec, int heightMeasureSpec)
             if (ratioWidth == 0 || ratioHeight == 0)
             {
                 SetMeasuredDimension(width, height);
+                return;
             }
-            else
+
+            switch (aspect)
             {
-                if (width < (float)height * ratioWidth / (float)ratioHeight)
-                {
-                    SetMeasuredDimension(width, width * ratioHeight / ratioWidth);
-                }
-                else
-                {
-                    SetMeasuredDimension(height * ratioWidth / ratioHeight, height);
-                }
+                case Aspect.AspectFit:
+                    if (width < (float)height * ratioWidth / (float)ratioHeight)
```

Comment:
I think might be better to force height, width, ratioWidth and rationHeight to be floats, to avoid any issues with calculations surrounding casting.

Reply:
do you mean to cast those to a float? they're returned as int from the Android APIs and casting is results in a redundant cast warning

Reply:
right now the code is a mess with various casts and almost impossible to verify there's not calculation errors due to rounding. The incremental cost of casting once at beginning v's the unreadable nature of the code is well worth it

Reply:
right got what you mean now, cast the initial result so we don't have to do it for the rest